tests/test_setup_config.py

```
@@ -1367,6 +1367,74 @@ def test_get_prerelease_patterns_selected_assets_key():
+        assert setup_config.WINDOWS_MODULES_AVAILABLE is True
+
+
+def test_windows_modules_import_failure(mocker, capfd):
+    """Test Windows modules import failure with helpful message."""
+    # Mock platform.system to return Windows
+    mocker.patch("platform.system", return_value="Windows")
+
+    # Mock failed imports by raising ImportError
+    def mock_import(name, *args, **kwargs):
+        if name in ["winshell", "win32com.shell"]:
+            raise ImportError(f"No module named '{name}'")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)